### ISSUE TITLE
Formatted text such as tables and formulas do not work in mobile

### ISSUE THREAD

**Issue by cjbirk** [mattermost/mattermost-mobile#2069]
#### Summary
Formatted text such as tables and formulas do not work in mobile

#### Environment Information

- Device Name: iPhone X
- OS Version: 11.4.1 (15G77)
- Mattermost App Version: 1.11.0 (Build 133)
- Mattermost Server Version: 5.2.0 (Build 5.2.1)

#### Steps to reproduce
Using the desktop or webapp: Paste formatted text, such as a table or formula:

```
#### Test results for July 27th, 2017
@channel please review failed tests.

  | Component  | Tests Run   | Tests Failed                                   |
  |:-----------|:-----------:|:-----------------------------------------------|
  | Server     | 948         | :white_check_mark: 0                           |
  | Web Client | 123         | :warning: 2 [(see details)](http://linktologs) |
  | iOS Client | 78          | :warning: 3 [(see details)](http://linktologs) |
```

and view it on your mobile device. The text is not formatted.

#### Expected behavior
The formatted text should render just as it does in the desktop or webapp.

#### Observed behavior
I observe this behavior both on iOS and Android running the latest version of the app and server as of 09/02/2018. I've observed this behavior since I've started using Mattermost (4.x release for server).
Android:
![android_formatted_text_broken](https://user-images.githubusercontent.com/2327149/44958728-2ef4b180-aeb2-11e8-9df1-cd8fe0a432b6.png)

iOS:
![ios_formatted_text_broken](https://user-images.githubusercontent.com/2327149/44958731-47fd6280-aeb2-11e8-841f-7b93e719725b.jpg)

#### Possible fixes
I am new to the project, but would be willing to work on this if given some guidance.

**Comment by kongr45gpen** [comment]
After some conversation on [~developer-toolkit](https://pre-release.mattermost.com/core/pl/sqdxs4nde7r98cxkgpdkrbxtjy), it looks like the problem comes from *spaces at the start of each line* of the table.

Here's a simple failing example:
```markdown
 | A  | B  | C  |
 |:---|:---|:---|
 | 1  | Name | Toast |
 | 2  | Name | Server |
```

**Comment by enahum** [comment]
@hmhealey, @saturninoabril I think there is a small difference to render tables between the webapp's markdown and mobile's commonmark libraries, can you please take a look?

**Comment by hmhealey** [comment]
Nice find, and great investigation into it being caused by the whitespace at the beginning of the line.

I filed a Jira ticket filed to look into this more ([link](https://mattermost.atlassian.net/browse/MM-11955)), but if anyone is interested in helping fix this, I can give some pointers. The problem code is likely [here in our commonmark.js fork](https://github.com/mattermost/commonmark.js/blob/master/lib/blocks.js#L638). It's likely caused by returning 0 when `parser.indented` is true or by one of `reTableDelimiter` or `reTableRow` not supporting spaces at the start of the line

**Comment by cjbirk** [comment]
@hmhealey Would you call this an 'introductory' level fix? I'm not a javascript expert but would be willing to take a stab if it is something you would consider beginner level.

**Comment by enahum** [comment]
@cjbirk there is already a fix for this https://github.com/mattermost/mattermost-mobile/pull/2114 but I guess it will be included in 1.13 scheduled to be released Oct 16th

**Comment by amyblais** [comment]
Closing this Issue as the Jira ticket has been resolved and tested internally - it will be available in v1.13 release.
